ports/osg/portfile.cmake

```
@@ -10,49 +8,75 @@ vcpkg_from_github(
 	HEAD_REF master
     PATCHES
         collada.patch
+        static.patch
+        fix-sdl.patch
+        fix-example-application.patch
+        disable-present3d-staticview-in-linux.patch #Due to some link error we cannot solve yet, disable them in linux.
 )
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
     set(OSG_DYNAMIC OFF)
 else()
     set(OSG_DYNAMIC ON)
 endif()
+
 file(REMOVE ${SOURCE_PATH}/CMakeModules/FindSDL2.cmake)
+
+set(OSG_USE_UTF8_FILENAME ON)
+if (NOT VCPKG_TARGET_IS_WINDOWS)
+    message("Build osg requires gcc with version higher than 4.7.")
+    # Enable OSG_USE_UTF8_FILENAME will call some windows-only functions.
+    set(OSG_USE_UTF8_FILENAME OFF)
+endif()
+
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     OPTIONS
-        -DOSG_USE_UTF8_FILENAME=ON
+        -DOSG_USE_UTF8_FILENAME=${OSG_USE_UTF8_FILENAME}
+        -DDYNAMIC_OPENSCENEGRAPH=${OSG_DYNAMIC}
+        -DDYNAMIC_OPENTHREADS=${OSG_DYNAMIC}
+        -DBUILD_OSG_EXAMPLES=ON
+        -DBUILD_OSG_APPLICATIONS=ON
 )
 
 vcpkg_install_cmake()
 
+vcpkg_copy_pdbs()
+
 # handle osg tools and plugins
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
-set(OSG_TOOL_PATH ${CURRENT_PACKAGES_DIR}/tools/osg)
+set(OSG_TOOL_PATH ${CURRENT_PACKAGES_DIR}/tools/${PORT})
 file(MAKE_DIRECTORY ${OSG_TOOL_PATH})
 
-file(GLOB OSG_TOOLS ${CURRENT_PACKAGES_DIR}/bin/*.exe)
+if (VCPKG_TARGET_IS_WINDOWS)
+    set(EXECUTE_SUFFIX ".exe")
+    set(DYNAMIC_SUFFIX ".dll")
+else()
+    set(EXECUTE_SUFFIX "")
+    set(DYNAMIC_SUFFIX ".so")
+endif()
+
+file(GLOB OSG_TOOLS ${CURRENT_PACKAGES_DIR}/bin/*${EXECUTE_SUFFIX})
 file(COPY ${OSG_TOOLS} DESTINATION ${OSG_TOOL_PATH})
 file(REMOVE_RECURSE ${OSG_TOOLS})
-file(GLOB OSG_TOOLS_DBG ${CURRENT_PACKAGES_DIR}/debug/bin/*.exe)
+file(GLOB OSG_TOOLS_DBG ${CURRENT_PACKAGES_DIR}/debug/bin/*${EXECUTE_SUFFIX})
 file(REMOVE_RECURSE ${OSG_TOOLS_DBG})
 
-file(GLOB OSG_PLUGINS_DBG ${CURRENT_PACKAGES_DIR}/debug/bin/osgPlugins-3.6.4/*.dll)
-file(COPY ${OSG_PLUGINS_DBG} DESTINATION ${CURRENT_PACKAGES_DIR}/debug/tools/osg/osgPlugins-3.6.4)
-file(GLOB OSG_PLUGINS_REL ${CURRENT_PACKAGES_DIR}/bin/osgPlugins-3.6.4/*.dll)
+file(GLOB OSG_PLUGINS_DBG ${CURRENT_PACKAGES_DIR}/debug/bin/osgPlugins-3.6.4/*${DYNAMIC_SUFFIX})
```

I think the portfile should parameterize the version number, since it's used pretty extensively.

```suggestion
set(OSG_VERSION "3.6.4")
```